dmutils/content_loader.py

```diff
@@ -156,8 +156,11 @@ def get_field_names(self):
     def _has_pricing_type(self):
         return any(self._is_pricing_type(q) for q in self.get_question_ids())
 
-    def get_question_ids(self):
-        return [question['id'] for question in self.questions]
+    def get_question_ids(self, type=None):
+        return [
+            question['id'] for question in self.questions
+            if type is None or question.get('type') == type
```

Comment:
~~Does every question definitely have a type?  If not then is there a sensible default that could be added to the `.get()` here?~~

OK, yep questions must have a type or the frontend wouldn't be able to render them and all sorts of things would break.